polars/polars-lazy/polars-plan/src/dsl/function_expr/strings.rs

```diff
@@ -142,34 +142,34 @@ pub(super) fn rjust(s: &Series, width: usize, fillchar: char) -> PolarsResult<Se
     Ok(ca.rjust(width, fillchar).into_series())
 }
 
-pub(super) fn strip(s: &Series, matches: Option<char>) -> PolarsResult<Series> {
+pub(super) fn strip(s: &Series, matches: &Option<String>) -> PolarsResult<Series> {
     let ca = s.utf8()?;
     if let Some(matches) = matches {
         Ok(ca
-            .apply(|s| Cow::Borrowed(s.trim_matches(matches)))
+            .apply(|s| Cow::Borrowed(s.trim_matches(|c| matches.contains(c))))
             .into_series())
     } else {
         Ok(ca.apply(|s| Cow::Borrowed(s.trim())).into_series())
     }
 }
 
-pub(super) fn lstrip(s: &Series, matches: Option<char>) -> PolarsResult<Series> {
+pub(super) fn lstrip(s: &Series, matches: &Option<String>) -> PolarsResult<Series> {
     let ca = s.utf8()?;
 
     if let Some(matches) = matches {
         Ok(ca
-            .apply(|s| Cow::Borrowed(s.trim_start_matches(matches)))
+            .apply(|s| Cow::Borrowed(s.trim_start_matches(|c| matches.contains(c))))
             .into_series())
     } else {
         Ok(ca.apply(|s| Cow::Borrowed(s.trim_start())).into_series())
     }
 }
 
-pub(super) fn rstrip(s: &Series, matches: Option<char>) -> PolarsResult<Series> {
+pub(super) fn rstrip(s: &Series, matches: &Option<String>) -> PolarsResult<Series> {
     let ca = s.utf8()?;
     if let Some(matches) = matches {
         Ok(ca
-            .apply(|s| Cow::Borrowed(s.trim_end_matches(matches)))
+            .apply(|s| Cow::Borrowed(s.trim_end_matches(|c| matches.contains(c))))
```

Comment:
This is a great solution for mutliple chars (e.g. `String`). However I would also like an extra branch for the case when the `String` contains of a single `char`, as this will likely be much faster.

Reply:
Good suggestion, will implement this! 👍

Reply:
I implemented the fast path (see current state of the PR), but I don't find a meaningful difference in my simple benchmark:

![image](https://user-images.githubusercontent.com/3502351/210082577-0af8c9f9-31cb-41ea-bf5f-754e2d25168c.png)

Maybe the fast path is not worth it? Or is my benchmarking approach wrong?

Reply:
I think it better shows on a few very long strings. 

We also need a release compilation as a development build is not representative of optimizations done, so they could be the same in a dev build. 

I think it is fine as is, but if you want to benchmark, don't forget to comment out fat linking and single thread compilation in the `Cargo.toml` otherwise you'll have to wait a looong time. :)